.github/workflows/distribution.yml

```diff
@@ -154,3 +154,40 @@ jobs:
         with:
           user: __token__
           password: ${{ secrets.pypi_password }}
+
+  pyg4ometry-docker:
+    name: Push pyg4ometry docker image
+    needs: [upload_all]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Dump job context
+        env:
+          JOB_CONTEXT: ${{ toJson(job) }}
+        run: echo "$JOB_CONTEXT"
+      - name: Check username
+        run: |
+          echo ${{ vars.BOOGERT_DOCKER_USERNAME }}
+
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          Username: ${{ vars.BOOGERT_DOCKER_USERNAME }}
+          password: ${{ secrets.BOOGERT_DOCKER_PASSWORD }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: g4edge/pyg4ometry
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: ./.github/docker
+          file: ./.github/docker/ubuntu22-pyg4ometry
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
This workflow has a `workflow_dispatch` option, which is very useful to test it without making a release. We don't want to actually push the image in those cases though. We have to add some if statement to this step.